aiofreepybox/api/player.py

```diff
@@ -0,0 +1,227 @@
+    media_control_stream_args = {"stream": media_control_stream, "type": "stream"}
+    media_control_track_args = {"track_id": 0, "type": "track_id"}
+    media_control_command = {
+        "play",
+        "pause",
+        "play_pause",
+        "stop",
+        "next",
+        "prev",
+        "seek_forward",
+        "seek_backward",
+        "seek_to",
+        "repeat_all",
+        "repeat_one",
+        "repeat_off",
+        "repeat_toggle",
+        "shuffle_on",
+        "shuffle_off",
+        "shuffle_toggle",
+        "record",
+        "record_stop",
+        "select_audio_track",
+        "select_srt_track",
+        "select_stream",
+    }
+    media_control_data_schema = {"args": media_control_stream_args, "cmd": "pause"}
+
+    async def _get_d_p_id(self) -> Optional[int]:
+        """
+        Get default player id
+        """
+
+        player = await self.get_players()
+        if player is not None:
+            return player[0]["id"]
+        return None
+
+    async def get_player_status(
+        self, player_id: Optional[int] = None
+    ) -> Optional[Dict[str, Any]]:
+        """
+        Get player status
+
+        player_id : `int` , optional
+            , Default to `None`
+        """
+
+        if player_id is None:
+            player_id = await self._get_d_p_id()
+        if player_id is not None:
+            return await self._access.get(
+                f"player/{player_id}/api/{self._player_api_version}/status/"
+            )
+        return None
+
+    async def get_player_volume(
+        self, player_id: Optional[int] = None
+    ) -> Optional[Dict[str, Any]]:
+        """
+        Get player volume
+
+        player_id : `int` , optional
+            , Default to `None`
+        """
+
+        if player_id is None:
+            player_id = await self._get_d_p_id()
+        if player_id is not None:
+            return await self._access.get(
+                f"player/{player_id}/api/{self._player_api_version}/control/volume"
+            )
+        return None
+
+    async def get_players(self) -> Optional[List[Dict[str, Any]]]:
+        """
+        Get players
+        """
+        return await self._access.get("player")
+
+    async def mute_player_switch(
+        self, enabled: Optional[bool] = None, player_id: Optional[int] = None
+    ) -> Optional[bool]:
+        """
+        Mute switch
+
+        enabled : `bool`, optional
+            , Default to `None`
+        player_id : `int`, optional
+            , Default to `None`
+        """
+
+        if player_id is None:
+            player_id = await self._get_d_p_id()
+        if player_id is not None:
+            if enabled is not None:
+                player_mute_data = {"mute": enabled}
+                await self.set_player_volume(player_mute_data, player_id)
+            player_mute = await self.get_player_volume(player_id)
+            if player_mute is not None:
+                return player_mute["mute"]
+        return None
```

Comment:
I wonder if it wouldn't be clearer to split this function into one function to set the mute status (with a not-optional enabled parameter), one to toggle the mute status (without enabled parameter) and one to get the mute status. I'm afraid the current version can be a bit confusing, with the function acting sometimes as both a setter and a getter, and sometimes only as a getter. What do you think?

Reply:
Oh but I'm seeing there is already a set_player_volume (and a user-friendly update_player_volume) and a get_player_volume, so I don't really understand what it the use of this function?

Reply:
Indeed, call `update_player_volume(mute=enabled, player_id=player_id)` instead.

Then `player_mute = await self.get_player_volume(player_id)`, etc.